mokuro/script.js

```diff
@@ -119,6 +121,39 @@ function updateProperties() {
         r.style.setProperty('--textBoxDisplay', 'none');
     }
 
+    if (state.toggleOCRTextBoxes) {
+        // Force a hovered state on the clicked .textBox, until the .textBox is clicked off of.
+        let textBoxes = document.querySelectorAll('.textBox');
+        for (let i = 0; i < textBoxes.length; i++) {
+            textBoxes[i].addEventListener('click', function (e) {
+                if (state.toggleOCRTextBoxes) {
+                    if (this.classList.contains('hovered')) {
+                        this.classList.remove('hovered');
+                    } else {
+                        this.classList.add('hovered');
+                        // Remove hovered state from all other .textBoxes
+                        for (let j = 0; j < textBoxes.length; j++) {
+                            if (i !== j) {
+                                textBoxes[j].classList.remove('hovered');
+                            }
+                        }
+                    }
+                }
+            });
+        }
+        // When clicking off of a .textBox, remove the hovered state.
+        document.addEventListener('click', function (e) {
+            if (state.toggleOCRTextBoxes) {
+                if (e.target.closest('.textBox') === null) {
+                    let textBoxes = document.querySelectorAll('.textBox');
+                    for (let i = 0; i < textBoxes.length; i++) {
+                        textBoxes[i].classList.remove('hovered');
+                    }
+                }
+            }
+        });
```

Comment:
This adds a new pair of event listeners without removing the old ones every time you call `updateProperties` as long as `state.toggleOCRTextBoxes` is true, potentially creating a large number of event listeners.

By the way this also leads to a funny bug - the consequent event listeners add and remove the hovered class in turns, so when the number of event listeners is even, they cancel each other out and it stops working.

My suggestion is to add the event listeners once on document load - since they check `state.toggleOCRTextBoxes`, they'd just do nothing if it's false.